web/src/message_events.js

```diff
@@ -626,6 +626,10 @@ export function update_messages(events) {
                 //       with data fetched from the server (which is already updated)
                 //       when we move to new narrow and what data is locally available.
                 if (changed_narrow) {
+                    // Remove outdated cached data to avoid repopulating from it.
+                    // We are yet to update the cached message list data for
+                    // the moved topics.
+                    message_list_data_cache.remove(new_filter);
```

Comment:
I see how this is a local improvement, but it seems like this shouldn't be necessary if we just manage order-of-operations better?

Reply:
Agreed, but I think it is better left as an optimization to be done later after we are out of woods with major bugs. Added a TODO comment for that here.